tests/bluetooth/tester/src/btp_micp.c

```diff
@@ -0,0 +1,534 @@
+		if (bt_aics_gain_setting_get(micp_included.aics[i]) != 0) {
+			return BTP_STATUS_FAILED;
+		}
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t micp_mute(const void *cmd, uint16_t cmd_len,
+			 void *rsp, uint16_t *rsp_len)
+{
+	int rc;
+
+	LOG_DBG("");
+
+	rc = bt_micp_mic_ctlr_mute(mic_ctlr);
+	if (rc != 0) {
+		return BTP_STATUS_FAILED;
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static uint8_t micp_aics_man_gain(const void *cmd, uint16_t cmd_len,
+				  void *rsp, uint16_t *rsp_len)
+{
+	LOG_DBG("AICS set manual gain mode");
+
+	for (int i = 0; i < micp_included.aics_cnt; i++) {
+		if (bt_aics_manual_gain_set(micp_included.aics[i]) != 0) {
+			return BTP_STATUS_FAILED;
+		}
+	}
+
+	return BTP_STATUS_SUCCESS;
+
+}
+
+static uint8_t micp_aics_auto_gain(const void *cmd, uint16_t cmd_len,
+				   void *rsp, uint16_t *rsp_len)
+{
+	LOG_DBG("AICS set automatic gain mode");
+
+	for (int i = 0; i < micp_included.aics_cnt; i++) {
+		if (bt_aics_automatic_gain_set(micp_included.aics[i]) != 0) {
+			return BTP_STATUS_FAILED;
+		}
+	}
+
+	return BTP_STATUS_SUCCESS;
+}
+
+static const struct btp_handler micp_handlers[] = {
+	{
+		.opcode = BTP_MICP_READ_SUPPORTED_COMMANDS,
+		.index = BTP_INDEX_NONE,
+		.expect_len = 0,
+		.func = micp_supported_commands,
+	},
+	{
+		.opcode = BTP_MICP_DISCOVER,
+		.expect_len = sizeof(struct btp_micp_discover_cmd),
+		.func = micp_discover,
+	},
+	{
+		.opcode = BTP_MICP_SET_GAIN,
+		.expect_len = sizeof(struct btp_micp_set_gain_cmd),
+		.func = micp_set_gain,
+	},
+	{
+		.opcode = BTP_MICP_AICS_STATE,
+		.expect_len = sizeof(struct btp_micp_aics_state_cmd),
+		.func = micp_aics_state,
+	},
+	{
+		.opcode = BTP_MICP_MUTE_READ,
+		.expect_len = sizeof(struct btp_micp_mute_read_cmd),
+		.func = micp_mute_read,
+	},
+	{
+		.opcode = BTP_MICP_GAIN_SETTING_PROP,
+		.expect_len = sizeof(struct btp_micp_gain_setting_prop_cmd),
+		.func = micp_gain_setting_prop,
+	},
+	{
+		.opcode = BTP_MICP_MUTE,
+		.expect_len = sizeof(struct btp_micp_mute_cmd),
+		.func = micp_mute,
+	},
+	{
+		.opcode = BTP_MICP_AICS_TYPE,
+		.expect_len = sizeof(struct btp_micp_aics_type_cmd),
+		.func = micp_aics_type,
+	},
+	{
+		.opcode = BTP_MICP_AICS_STATUS,
+		.expect_len = sizeof(struct btp_micp_aics_status_cmd),
+		.func = micp_aics_status,
+	},
+	{
+		.opcode = BTP_MICP_AICS_UNMUTE,
+		.expect_len = sizeof(struct btp_micp_aics_unmute_cmd),
+		.func = micp_aics_unmute,
+	},
+	{
+		.opcode = BTP_MICP_AICS_MUTE,
+		.expect_len = sizeof(struct btp_micp_aics_mute_cmd),
+		.func = micp_aics_mute,
+	},
+	{
+		.opcode = BTP_MICP_AICS_MAN_GAIN,
+		.expect_len = sizeof(struct btp_micp_aics_manual_gain_cmd),
+		.func = micp_aics_man_gain,
+	},
+	{
+		.opcode = BTP_MICP_AICS_AUTO_GAIN,
+		.expect_len = sizeof(struct btp_micp_aics_auto_gain_cmd),
+		.func = micp_aics_auto_gain,
+	},
+};
+
+uint8_t tester_init_micp(void)
+{
+	int err;
+
+	err = bt_micp_mic_ctlr_cb_register(&micp_cbs);
+	if (err != 0) {
+		LOG_DBG("Failed to register callbacks: %d", err);
```

Comment:
return BTP_STATUS_FAILED;

we don't want to initialize service  if those failed